### PR TITLE
v1.14 Backports 2023-12-07

### PR DESCRIPTION
**Pull request by nbusseneau**
* [x] #27863 (@tommyp1ckles)
 * [x] #29623 (@pchaigno)
 * [x] #29653 (@brb)

Once this PR is merged, a GitHub action will update the labels of these PRs:
```upstream-prs
 27863 29623 29653
```

### CODE REVIEW

**Comment by nbusseneau** [comment]
@tommyp1ckles @pchaigno @brb Conflicts were hit while backporting your commits, please check backporter's notes for details :)

**Comment by nbusseneau** [comment]
@tommyp1ckles In particular, does it make sense to even make changes in `upgrade.rst` here? I was unsure, so decided to roll with it, but let me know... 🤔

**Comment by nbusseneau** [comment]
/test-backport-1.14

**Review by pchaigno — APPROVED** [review]

**Review by ti-mo — APPROVED** [review]

**Comment by brb** [comment]
The ci-ipsec-upgrade seems to be failing due to the same reason - https://github.com/cilium/cilium/issues/29351. cc @qmonnet

**Comment by qmonnet** [comment]
> The ci-ipsec-upgrade seems to be failing due to the same reason - #29351. cc @qmonnet

The failure on kernel 5.4 is #29351 indeed, occurring most frequently with 5.10 but also sometimes with 5.4. I'm working on that one but haven't found the root cause yet.

The failure with bpf-next is another recurrent failure on the workflow, I'm not sure we've got an open issue for this one (I'll check and create one if necessary).

The failure with 6.1 I've never seen before :thinking:

**Comment by brb** [comment]
Let's first merge https://github.com/cilium/cilium/pull/29742, and then see whether the issue got resolved.

**Comment by nbusseneau** [comment]
@qmonnet So my assumption is that we're safe to ignore the failures in `ci-ipsec-upgrade` because known failures + it's not marked as required, correct?

**Comment by qmonnet** [comment]
> @qmonnet So my assumption is that we're safe to ignore failures in `ci-ipsec-upgrade` because known failures + it's not marked as required, correct?

It's not marked as required so I suppose it's acceptable? But no, the failure on kernel 6.1 was not known to me.

Same for v1.13: I expect #29741/#29742 to address the failures on the 5.4 kernel jobs, not with 6.1.

**Review by tommyp1ckles — APPROVED** [review]

**Comment by giorio94** [comment]
Marking as blocked while IPSec the new failures in Cilium IPSec upgrade get investigated: https://github.com/cilium/cilium/pull/29702#issuecomment-1854585443

File: .github/workflows/tests-ipsec-upgrade.yaml [hunk 1]
```diff
@@ -97,6 +97,15 @@ jobs:
...
 
+          - config: '6.1'
+            # renovate: datasource=docker depName=quay.io/lvh-images/kind
+            kernel: '6.1-20231026.065108'
+            kube-proxy: 'iptables'
+            kpr: 'disabled'
+            tunnel: 'vxlan'
+            encryption: 'ipsec'
+            endpoint-routes: 'false'
+
```
Comment (brb) [inline .github/workflows/tests-ipsec-upgrade.yaml:109]: @nbusseneau Could you append the following to the commit which added this config?
```
# https://github.com/cilium/cilium/issues/29805#issuecomment-1855600171
ipv6: 'false'
```

**Comment by brb** [comment]
/test-backport-1.14

**Review by brb — APPROVED** [review]

**Comment by brb** [comment]
/test-backport-1.14

**Pull request merged**